lib/indexed_dao.es6.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+foam.CLASS({
+  package: 'org.chromium.apis.web',
+  name: 'AbstractFacetedIndexed',
+
+  documentation: `Abstract class for something that is faceted (stores a Class
+      in "of") and indexed according to "indices" property.`,
+
+  properties: [
+    {
+      class: 'Class',
+      name: 'of',
+    },
+    {
+      class: 'Array',
+      name: 'indices',
```

Comment:
Since this is actually holding a list of properties to index (and not indexes themselves) I'd rename it to avoid confusion with the MDAO's actual root index.